smcore/languages.cpp

```diff
@@ -0,0 +1,65 @@
+//******************************************************************************
+// Copyright (C) 1999 Jim Wanner and the SourceMonitor team.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+//******************************************************************************
+
+#pragma once
+
+#include "languages.h"
+
+namespace smos
+{
+    namespace smcore
+    {
+        //******************************************************************************
+        const char *langguagestrings[] = {
```

Comment:
Naming convention is out of whack! "langGuageStrings"? This should be a static class instance, with getters/setters.

Reply:
Uh, this is first a naming issue, should be "languageStrings". Then - a static class instance is accepted. But getters/setters? This property will reflect the enum smos::smcore::Languages::Type. These are the languages SourceMonitor could parse.